openstack/datadog_checks/openstack/openstack.py

```diff
@@ -95,6 +96,11 @@
     'ACTIVE'
 ]
 
+REMOVED_STATES = [
+    'DELETED',
+    'SHUTOFF'
```

Comment:
unsure `SHUTOFF` should lead to the the same actions as `REMOVED`; I see you use these states to clear out `servers` from the cache.... I would ask myself if it's worth it to keep `SHUTOFF` servers around in the cache. I don't have a strong opinion, but would like us to think about it. This would allow us to report a metric on number of stopped VMs, etc... or ultimately add some "sensors" for stopped VM's whatever they may be. Bear in mind that VMs might have storage, or networking or whatever, associated with them (and although we might not be reporting those metrics now, it' might be worth it or more future proof to keep as rich a server registry in cache as we can). Then again, a larger registry could also add a performance penalty.

Reply:
So I think that having this would let us add some future metrics about server types and what not, but I don't think having them in the cache is worthwhile for this performance based work. 

In terms of being future proof, I think having the cache setup as it currently is, is rather extensible for the future should we want to add in those metrics.